components/services/brave_wallet/public/cpp/brave_wallet_utils_service.cc

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/services/brave_wallet/public/cpp/brave_wallet_utils_service.h"
+
+#include <utility>
+
+#include "base/no_destructor.h"
+#include "build/build_config.h"
+
+#if BUILDFLAG(IS_IOS)
+#include "brave/components/services/brave_wallet/public/cpp/brave_wallet_utils_service_in_process_launcher.h"
+#else
+#include "brave/components/services/brave_wallet/content/brave_wallet_utils_service_launcher.h"
+#endif
+
+namespace brave_wallet {
+
+BraveWalletUtilsService::BraveWalletUtilsService() = default;
+BraveWalletUtilsService::~BraveWalletUtilsService() = default;
+
+// static
+BraveWalletUtilsService* BraveWalletUtilsService::GetInstance() {
+  static base::NoDestructor<BraveWalletUtilsService> service;
+  return service.get();
+}
+
+void BraveWalletUtilsService::CreateZCashDecoder(
+    mojo::PendingAssociatedReceiver<mojom::ZCashDecoder> receiver,
+    base::OnceCallback<void()> callback) {
+  MaybeLaunchService();
+  brave_wallet_utils_service_->CreateZCashDecoderService(std::move(receiver),
+                                                         std::move(callback));
+}
+
+void BraveWalletUtilsService::MaybeLaunchService() {
+#if BUILDFLAG(IS_IOS)
+  LaunchInProcessBraveWalletUtilsService(
+      brave_wallet_utils_service_.BindNewPipeAndPassReceiver());
+  brave_wallet_utils_service_.reset_on_disconnect();
+#else
+  LaunchBraveWalletUtilsService(
+      brave_wallet_utils_service_.BindNewPipeAndPassReceiver());
+  brave_wallet_utils_service_.reset_on_disconnect();
+  brave_wallet_utils_service_.reset_on_idle_timeout(base::Minutes(5));
```

Comment:
5 mins seems a bit long for an idle process? What's our use case here and can it be shorter?

Reply:
Such timeout should somehow relate to wallet's auto-lock timer(maybe just it's default value). Also makes sense to shutdown process when wallet gets locked.

Reply:
Logged follow-up issue https://github.com/brave/brave-browser/issues/35568